### DISCUSSION TITLE
A simple JSON formatter

### DISCUSSION THREAD

**Discussion by blackhaj** [johnlindquist/kit#664] [Share]
Hey folks - I made a little JSON formatter (my first attempt at a script)

It does the following:

Grabs the latest item on your clipboard
Formats it using prettier (this will have to install on your first time running the script)
Copies the formatted JSON to your clipboard
Super simple but hopefully useful.

[Open json-formatter in Script Kit](https://scriptkit.com/api/new?name=json-formatter&url=https://gist.githubusercontent.com/blackhaj/6651f50c6755db63e25f09f5cb9fbcaf/raw/c8a05d7266d46795d341dd4772bf9a3ad8e000c2/json-formatter.js")

```js
import '@johnlindquist/kit';

const prettier = await npm('prettier');

// Name: json-formatter
// Author: Henry Black
// Twitter: @hajblack

const unformattedJson = await paste();

let result;

try {
  result = await prettier.format(unformattedJson, {
    parser: 'json',
  });
  await copy(result);
} catch (err) {
  arg("Couldn't format that string: " + err);
}

```

Any feedback very welcome - this is my first script on ScriptKit so I would love any pointers.

**Comment by johnlindquist** [comment]
Looks great to me 👍

1. The `// Menu` metadata is "deprecated", I'll have to go remove that from the docs/older examples. `// Name` is now the "recommended", but either works great.

2. When sharing, if you press "right arrow" on your keyboard with a script focused, it will bring up the alternate actions menu:
![CleanShot 2021-12-20 at 11 29 34@2x](https://user-images.githubusercontent.com/36073/146815411-34e87726-55b4-4002-8b51-71da68865851.png)

"Left arrow" will take you back to the main menu
<img width="802" alt="CleanShot 2021-12-20 at 11 30 07@2x" src="https://user-images.githubusercontent.com/36073/146815485-07381e23-c9b1-49bc-a9ba-5e933cd77098.png">

Using the "Share as GitHub Discussion" will create nicely formatted post along with "install link" for the script. Here's an example:


[Open clipboard-history in Script Kit](https://scriptkit.com/api/new?name=clipboard-history&url=https://gist.githubusercontent.com/johnlindquist/5fc4de5137d8b6b2bf28d7ee451d3998/raw/eead9d1ba3fa2b87581a6dd062d93672fd5ecb98/clipboard-history.js")

```js
let history = await getClipboardHistory()
let historyWithPreviews = history.map(item => {
  item.preview = () => `<div class="text-xs p-5 font-mono">${item.value}</div>`
  return item
})
let value = await arg("Hit enter to paste", historyWithPreviews)

await setSelectedText(value)

```

3. Clipboard History

There's some basic apis for getting other items from the clipboard (instead of just the most recent) as shown in the script above. Feel free to play with it.

**Reply by blackhaj** [reply]
Thanks so much for the clear explanation, John. I will resubmit this in a moment following the steps you mentioned

**Reply by johnlindquist** [reply]
Thanks, in the future, I'll find a way to "filter" discussions from the site/hot instead of requiring deleting them. Just hadn't really thought about it until now
